src/utils/processes/utils.nf

```diff
@@ -474,6 +474,45 @@ process SC__PUBLISH {
         """
 }
 
+process MAKE_UNIQUE_FILENAME {
+
+    label 'compute_resources__minimal'
+    
+    input:
+        tuple \
+            val(tag), \
+            path(f), \
+            val(stashedParams)
+        val(fileOutputSuffix)
+        val(toolName)
+        val(isParameterExplorationModeOn)
+
+    output:
+        tuple \
+            val(tag), \
+            path(outputFileName), \
+            val(stashedParams)
+
+    script:
+        outputFileName = getOutputFileName(
+            params,
+            tag,
+            f,
+            fileOutputSuffix,
+            isParameterExplorationModeOn,
+            stashedParams
+        )
+        /* avoid cases where the input and output files have identical names:
+           Move the input file to a unique name, then create a link to
+           the input file */
+        """
+        mv $f tmp
+        if [ ! -f ${outputFileName} ]; then
+            ln -L tmp "${outputFileName}"
+        fi
+        """
+}
+
```

Comment:
Didn't we discuss using the `stageAs` option from nextflow for this kind of issue? It seems redundant to use a link and all the moving when nextflow is able to do this. I know that some functions use the name of the original file to determine some information, but then it seems that maybe this could just be passed as an argument to those processes.

Reply:
So you would remove this NXF process and use `stageAs` in the process where the output should be `groupTuble`d ?